DataFormats/ForwardDetId/interface/BTLDetId.h

```diff
@@ -50,18 +75,34 @@ class BTLDetId : public MTDDetId {
       ( module& kBTLmoduleMask ) << kBTLmoduleOffset |
       ( modtyp& kBTLmodTypeMask ) << kBTLmodTypeOffset |
       ( (crystal-1)& kBTLCrystalMask ) << kBTLCrystalOffset ; 
-}
+  }
+  
+  // ---------- Common methods ----------
+  
+  /** Returns BTL module number. */
+  inline int module() const { return (id_>>kBTLmoduleOffset)&kBTLmoduleMask; }
+  
+  /** Returns BTL crystal type number. */
+  inline int modType() const { return (id_>>kBTLmodTypeOffset)&kBTLmodTypeMask; }
+  
+  /** Returns BTL crystal number. */
+  inline int crystal() const { return ((id_>>kBTLCrystalOffset)&kBTLCrystalMask) + 1; }
+
+  /** Returns BTL iphi index for crystal according to type tile or bar */
+  int iphi( CrysLayout lay ) const ;
```

Comment:
is there a way the DetId could know if it is a tile or bar, so we don't have to rely on downstream code calling the correct version of the function?

Reply:
not really in the numbering scheme I have drafted this spring. With this trick I make the numbering scheme code invariant wrt the scenario. Of course this is just a temporary approach that should be removed as soon as the scenario is finally settled.